valhalla/thor/dijkstras.h

```diff
@@ -0,0 +1,273 @@
+   * @returns                  a pair with the first being the epoch seconds for the date time at that
+   *                           timezone and the second being the ordinal second from the beginning of
+   *                           the week
+   */
+  std::pair<uint64_t, uint32_t>
+  SetTime(google::protobuf::RepeatedPtrField<valhalla::Location>& locations,
+          const baldr::GraphId& node_id,
+          baldr::GraphReader& reader);
+
+  /**
+   * Expand from the node along the forward search path.
+   * @param graphreader  Graph reader.
+   * @param node Graph Id of the node to expand.
+   * @param pred Edge label of the predecessor edge leading to the node.
+   * @param pred_idx Index in the edge label list of the predecessor edge.
+   * @param from_transition Boolean indicating if this expansion is from a transition edge.
+   * @param localtime Current local time.  Seconds since epoch.
+   * @param seconds_of_week For time dependent Dijkstrass this allows lookup of predicted traffic.
+   */
+  void ExpandForward(baldr::GraphReader& graphreader,
+                     const baldr::GraphId& node,
+                     const sif::EdgeLabel& pred,
+                     const uint32_t pred_idx,
+                     const bool from_transition,
+                     uint64_t localtime,
+                     int32_t seconds_of_week);
+
+  /**
+   * Expand from the node along the reverse search path.
+   * @param graphreader  Graph reader.
+   * @param node Graph Id of the node to expand.
+   * @param pred Edge label of the predecessor edge leading to the node.
+   * @param pred_idx Index in the edge label list of the predecessor edge.
+   * @param from_transition Boolean indicating if this expansion is from a transition edge.
+   * @param localtime Current local time.  Seconds since epoch.
+   * @param seconds_of_week For time dependent Dijkstrass this allows lookup of predicted traffic.
+   */
+  void ExpandReverse(baldr::GraphReader& graphreader,
+                     const baldr::GraphId& node,
+                     const sif::BDEdgeLabel& pred,
+                     const uint32_t pred_idx,
+                     const baldr::DirectedEdge* opp_pred_edge,
+                     const bool from_transition,
+                     uint64_t localtime,
+                     int32_t seconds_of_week);
+
+  /**
+   * Expand from the node using multimodal algorithm.
+   * @param graphreader  Graph reader.
+   * @param node Graph Id of the node to expand.
+   * @param pred Edge label of the predecessor edge leading to the node.
+   * @param pred_idx Index in the edge label list of the predecessor edge.
+   * @param from_transition Boolean indicating if this expansion is from a transition edge.
+   * @param pc Pedestrian costing.
+   * @param tc Transit costing.
+   * @param mode_costing Array of all costing models.
+   */
+  void ExpandForwardMM(baldr::GraphReader& graphreader,
```

Comment:
Consistency: Can we use full form for `MM` - `MultiModal` (like in `ComputeMulitModal`)?